src/core/src/package_managers/TdnfPackageManager.py

```diff
@@ -0,0 +1,458 @@
+        #
+        # Total installed size:  36.26M
+        # Total download size:  10.62M
+        # Error(1032) : Operation aborted.
+        dependencies = []
+        package_arch_to_look_for = ["x86_64", "noarch", "i686", "aarch64"]  # if this is changed, review Constants
+
+        lines = output.strip().splitlines()
+
+        for line_index in range(0, len(lines)):
+            line = re.split(r'\s+', lines[line_index].strip())
+            dependent_package_name = ""
+
+            if self.is_valid_update(line, package_arch_to_look_for):
+                dependent_package_name = self.get_product_name_with_arch(line, package_arch_to_look_for)
+            else:
+                self.composite_logger.log_verbose("[TDNF] > Inapplicable line: " + str(line))
+                continue
+
+            if len(dependent_package_name) != 0 and dependent_package_name not in packages and dependent_package_name not in dependencies:
+                self.composite_logger.log_verbose("[TDNF] > Dependency detected: " + dependent_package_name)
+                dependencies.append(dependent_package_name)
+
+        return dependencies
+
+    def is_valid_update(self, package_details_in_output, package_arch_to_look_for):
+        # Verifies whether the line under consideration (i.e. package_details_in_output) contains relevant package details.
+        # package_details_in_output will be of the following format if it is valid
+        #   Sample package details in TDNF:
+        #   python3-libs                       x86_64                    3.12.3-5.azl3                      azurelinux-official-base   36.05M                      10.52M
+        return len(package_details_in_output) == 6 and self.is_arch_in_package_details(package_details_in_output[1], package_arch_to_look_for)
+
+    @staticmethod
+    def is_arch_in_package_details(package_detail, package_arch_to_look_for):
+        # Using a list comprehension to determine if chunk is a package
+        return len([p for p in package_arch_to_look_for if p in package_detail]) == 1
+
+    def get_dependent_list(self, packages):
+        """Returns dependent List for the list of packages"""
+        package_names = ""
+        for index, package in enumerate(packages):
+            if index != 0:
+                package_names += ' '
+            package_names += package
+
+        self.composite_logger.log_verbose("[TDNF] Resolving dependencies. [Command={0}]".format(str(self.single_package_upgrade_simulation_cmd + package_names)))
+        output = self.invoke_package_manager(self.single_package_upgrade_simulation_cmd + package_names)
+        dependencies = self.extract_dependencies(output, packages)
+        self.composite_logger.log_verbose("[TDNF] Resolved dependencies. [Packages={0}][DependencyCount={1}]".format(str(packages), len(dependencies)))
+        return dependencies
```

Comment:
This code is important for exclusions coverage. Specifically excluding a package because its dependency needs to be excluded. Please include in test case matrix.

Reply:
Tested, log in PR description